lib/matplotlib/widgets.py

```diff
@@ -1237,35 +1250,108 @@ def _get_data(self, event):
         ydata = min(y1, ydata)
         return xdata, ydata
 
+    def _clean_event(self, event):
+        """Clean up an event
+
+        Use prev event if there is no xdata
+        Limit the xdata and ydata to the axes limits
+        Set the prev event
+        """
+        if event.xdata is None:
+            event = self._prev_event
+        else:
+            event = copy.copy(event)
+        event.xdata, event.ydata = self._get_data(event)
+
+        self._prev_event = event
+        return event
+
     def press(self, event):
-        """Button press handler"""
+        """Button press handler and validator"""
         if not self.ignore(event):
-            self.eventpress = copy.copy(event)
-            self.eventpress.xdata, self.eventpress.ydata = (
-                self._get_data(event))
+            event = self._clean_event(event)
+            self.eventpress = event
+            self._prev_event = event
+            key = event.key or ''
+            key = key.replace('control', 'ctrl')
```

Comment:
We can change the default to 'control' and reverse them, if you like. 



—
Sent from Mailbox

On Mon, Aug 31, 2015 at 10:25 PM, Benjamin Root <notifications@github.com>
wrote:

>>          if not self.ignore(event):
>> -            self.eventpress = copy.copy(event)
>> -            self.eventpress.xdata, self.eventpress.ydata = (
>> -                self._get_data(event))
>> +            event = self._clean_event(event)
>> +            self.eventpress = event
>> +            self._prev_event = event
>> +            key = event.key or ''
>> +            key = key.replace('control', 'ctrl')
> although, reading further, I see the problem is if one is trying to do multiple things at once, isn't it?
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/matplotlib/matplotlib/pull/3937/files#r38382713